pkg/controller/controller.go

```diff
@@ -129,23 +129,34 @@ func (c *Controller) Run() error {
 			switch event.Type {
 			case "ADDED":
 				clusterSpec := &event.Object.Spec
-				stopC := make(chan struct{})
-				c.stopChMap[clusterName] = stopC
-
-				nc := cluster.New(c.KubeCli, clusterName, c.Namespace, clusterSpec, stopC, &c.waitCluster)
-				c.clusters[clusterName] = nc
-				analytics.ClusterCreated()
 
 				backup := clusterSpec.Backup
 				if backup != nil && backup.MaxSnapshot != 0 {
 					err := k8sutil.CreateBackupReplicaSetAndService(c.KubeCli, clusterName, c.Namespace, *backup)
 					if err != nil {
-						panic(err)
+						log.Errorf("cluster (%s) is dead due to failure to create backup: %v", clusterName, err)
+						continue
 					}
 				}
+
+				stopC := make(chan struct{})
+				c.stopChMap[clusterName] = stopC
+
+				nc := cluster.New(c.KubeCli, clusterName, c.Namespace, clusterSpec, stopC, &c.waitCluster)
+				c.clusters[clusterName] = nc
+
+				analytics.ClusterCreated()
 			case "MODIFIED":
+				if c.clusters[clusterName] == nil {
+					log.Warningf("cluster (%s) isn't alive, but received update", clusterName)
```

Comment:
I'm not sure.
It shouldn't happen previously.
But now since we allow to mark a cluster is dead but without forcing anything on user side, user can still change/delete the spec on his will.